LibrarySystem/Controllers/AuthController.cs

```diff
@@ -0,0 +1,375 @@
+
+            if (userStatus != false)
+            {              
+                var user = _userRepository.GetUserById(id);               
+           
+                if (user != null && id.ToString().Length < 10 )
+                {
+                    if (_authService.GetUserFromToken(accessToken) != user.Email)
+                    {
+                        var userDb = _userRepository.GetUserByEmail(user.Email);
+                        var authdb = _userRepository.GetAuthByEmail(user.Email);
+
+                        if (userDb == null)
+                        {
+                            return BadRequest(new { status = "error", message = "User deletion requires a Valid User ID." });
+                        }
+
+                        if (userDb.IsAdmin)
+                        {
+                            return BadRequest(new { status = "error", message = "Sorry...Admin Cannot delete an Admin" });
+                        }
+
+                        _userRepository.RemoveEntity<User>(userDb);
+                        _userRepository.RemoveEntity<Auth>(authdb);
+                        if (_userRepository.SaveChangers())
+                        {
+                            return Ok(new { status = "success", message = "User"+" "+id+" "+"Deleted successfully." });
+                        }
+
+                        return BadRequest(new { status = "error", message = "This user is not registered" });
+                    }
+                    return BadRequest(new { status = "error", message = "Unable to delete account. You cannot delete your own account." });
+                }
+                return BadRequest(new { status = "error", message = "User deletion requires a Valid User ID." });
+            }
+
+            return StatusCode(403, new { status = "forbidden", message = "You do not have permission to delete other users. " });
+        }
+
+
+
+
+
+        // search Users by Email, First name, Last Name, Phone Number,Gender
+        [HttpGet("GetUsers")]
+        public async Task<IActionResult> Search([FromQuery] string? criteria)
+        {
+            var accessToken = HttpContext.GetTokenAsync(JwtBearerDefaults.AuthenticationScheme, "access_token").Result;
+            var userStatus = _authService.GetStatusFromToken(accessToken);
+
+            if (userStatus)
+            {
+
+                try
+                {
+                    if (criteria != null)
+                    {
+                        // Deserialize the JSON criteria
+                        var searchCriteria = JsonConvert.DeserializeObject<User>(criteria);
+                        var users = _userRepository.SearchUsersSpefically(searchCriteria.FirstName, searchCriteria.LastName, searchCriteria.Email, searchCriteria.PhoneNumber, searchCriteria.Gender, searchCriteria.UserId);
+
+
+                        if (searchCriteria.FirstName.Length >= 3 || searchCriteria.LastName.Length >= 3 || searchCriteria.Email.Length >= 3
```

Comment:
shouldn't this check be done before the querying the DB

Reply:
Yes, fixed it